lib/pacemaker/pcmk_sched_native.c

```diff
@@ -1678,12 +1679,12 @@ native_internal_constraints(pe_resource_t * rsc, pe_working_set_t * data_set)
 
             if (pcmk_is_set(rsc->flags, pe_rsc_allow_remote_remotes)) {
                 score = 10000;    /* Highly preferred but not essential */
+                flags |= pcmk__coloc_bundle_connection;
```

Comment:
Should we make this independent of `pe_rsc_allow_remote_remotes`? I wonder if there's a situation where a bundle is running on a **cluster** node and its connection resource has to move to a different node for whatever reason. Would that cause the container to stop and move, and if so would maintaining colocation be worth the container move?

Reply:
Aside: this made me realize we could just check colocation->dependent for pe_rsc_allow_remote_remotes rather than add the pcmk__coloc_bundle_connection flag. So, I'll drop the bool-to-flags stuff and do that.

It's a good question whether we should do this for containers on cluster nodes.

Action initiation goes DC -> connection host -> container host, so the benefit of the colocation is making that communication more reliable and efficient. We're limiting this PR to active containers, so the colocation still has effect on initial placement, which is a good thing. The questions are:

* For a container on a remote node, is it worth moving the container if a cluster node becomes available?
* For a container on a cluster node, is it worth moving the container if the connection has to move (most likely due to too many failures)?

Before this PR, the answer was "yes" to both. It's not obvious what the ideal answer would be. Given that moving connections is lightweight with no service interruption but moving containers is heavyweight with service interruption, maybe "no" to both is better. The user can always clean up the remote connection to have it move back to the container. I'll check the effect on the regression tests to see if there is anything unexpected there.

Reply:
Sounds good. The bool-to-flags stuff may come in handy later so I'd stash it at least.

I lean toward the less disruptive option at the expense of a small increase in communication overhead, if it turns out not to cause any issues in regression testing.